# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/media-conduit/WebrtcMediaDataDecoderCodec.cpp
# Commit: d1535acfa6fe
# Full Hash: d1535acfa6fe14fcec6bba1fa08607c13e8c3cd3
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2018-11-09 10:01:40
# Regressor Bug: 1376873
# File Overlap Count: 1
# Description:
#   Bug 1376873 - WebrtcMediaDataDecoderCodec updates; r=pehrsons
#   
#   This makes the ImageBuffer class implement VideoFrameBuffer class and makes
#   it available in MediaPipeline. To avoid exposing MediaPipeline to a lot of
#   internal details of WebrtcMediaDataDecoder, ImageBuffer is moved to its
# ==============================================================================

diff -r 2fb84f9968af -r d1535acfa6fe media/webrtc/signaling/src/media-conduit/WebrtcMediaDataDecoderCodec.cpp
--- a/media/webrtc/signaling/src/media-conduit/WebrtcMediaDataDecoderCodec.cpp	Tue Aug 21 13:40:10 2018 -0400
+++ b/media/webrtc/signaling/src/media-conduit/WebrtcMediaDataDecoderCodec.cpp	Tue Aug 21 13:40:45 2018 -0400
@@ -9,7 +9,7 @@
 #include "VideoUtils.h"
 #include "mozilla/media/MediaUtils.h"
 #include "mozilla/layers/ImageBridgeChild.h"
-#include "webrtc/base/keep_ref_until_done.h"
+#include "webrtc/rtc_base/keep_ref_until_done.h"
 
 namespace mozilla {
 
@@ -172,36 +172,4 @@
   return OwnerThread()->IsCurrentThreadIn();
 }
 
-ImageBuffer::ImageBuffer(RefPtr<layers::Image>&& aImage)
-  : webrtc::NativeHandleBuffer(aImage,
-                               aImage->GetSize().width,
-                               aImage->GetSize().height)
-  , mImage(std::move(aImage))
-{
-}
-
-rtc::scoped_refptr<webrtc::VideoFrameBuffer>
-ImageBuffer::NativeToI420Buffer()
-{
-  RefPtr<layers::PlanarYCbCrImage> image = mImage->AsPlanarYCbCrImage();
-  if (!image) {
-    // TODO. YUV420 ReadBack, Image only provides a RGB readback.
-    return nullptr;
-  }
-  rtc::scoped_refptr<layers::PlanarYCbCrImage> refImage(image);
-  const layers::PlanarYCbCrData* data = image->GetData();
-  rtc::scoped_refptr<webrtc::VideoFrameBuffer> buf(
-    new rtc::RefCountedObject<webrtc::WrappedI420Buffer>(
-      data->mPicSize.width,
-      data->mPicSize.height,
-      data->mYChannel,
-      data->mYStride,
-      data->mCbChannel,
-      data->mCbCrStride,
-      data->mCrChannel,
-      data->mCbCrStride,
-      rtc::KeepRefUntilDone(refImage)));
-  return buf;
-}
-
 } // namespace mozilla